### PR TITLE
feat(core): dynamic prefix/suffix detection for model lookup

### PR DESCRIPTION
**Pull request by cantalupo555**
## Summary
- Replace hardcoded `STRIPPED_PREFIXES`, `TIER_SUFFIXES`, and `FALLBACK_SUFFIXES` with intelligent dynamic detection
- Automatically discovers routing prefixes and tier suffixes by progressively stripping segments and validating against the pricing database
- Addresses the concern raised in PR #91 about hardcoded values requiring code changes for each new routing plugin

## Changes
- Add `try_strip_unknown_suffix()` for progressive suffix stripping (up to 4 segments)
- Add `try_strip_unknown_prefix()` for progressive prefix stripping (up to 2 segments)
- Remove all hardcoded constant arrays
- Add `MIN_MODEL_NAME_LEN` guard (5 chars) to prevent false positives
- Add comprehensive integration tests for edge cases

## How It Works
1. Try direct lookup
2. Try stripping unknown suffixes progressively
3. Try stripping unknown prefixes progressively, also trying suffix stripping on each candidate

## Examples
Models like these now resolve automatically without any hardcoded knowledge:
- `antigravity-claude-opus-4-5-thinking-high` → `claude-opus-4-5`
- `myplugin-gpt-4o` → `gpt-4o`
- `router-v2-claude-sonnet-4-5` → `claude-sonnet-4-5`

## Testing
- 128 tests passing
- Build successful
- CLI verified working

Ref: junhoyeo/tokscale#91

### CODE REVIEW

File: packages/core/index.js [hunk 1]
```diff
@@ -536,17 +536,13 @@ if (!nativeBinding || process.env.NAPI_RS_FORCE_WASI) {
       wasiBindingError = err
     }
   }
-  if (!nativeBinding || process.env.NAPI_RS_FORCE_WASI) {
+  if (!nativeBinding) {
```
[the remark this answers is not in the document]
Comment (junhoyeo) [reply packages/core/index.js:539]: Won't fix
